### PR TITLE
migrate kubelet/server/server.go logs to structured logging

### PR DESCRIPTION
**Pull request by yxxhero**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
part of kubernetes/enhancements#1602

**Which issue(s) this PR fixes**:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
migrate kubelet/server/server.go logs to structured logging
```

### CODE REVIEW

**Comment by yxxhero** [comment]
/cc  @Random-Liu @yujuhong
Thanks very much.

File: pkg/kubelet/server/server.go [hunk 1]
```diff
@@ -145,7 +146,7 @@ func ListenAndServeKubeletServer(
 	enableDebuggingHandlers,
 	enableContentionProfiling,
 	enableSystemLogHandler bool) {
-	klog.Infof("Starting to listen on %s:%d", address, port)
```
Comment (yuzhiquan) [inline pkg/kubelet/server/server.go:148]: That's will be result in `Starting to listen on 127.0.0.1=8080`

File: pkg/kubelet/server/server.go [hunk 2]
```diff
@@ -159,23 +160,26 @@ func ListenAndServeKubeletServer(
...
-		klog.Fatal(s.ListenAndServe())
+		klog.ErrorS(s.ListenAndServe(), "Failed to listen and serve")
 	}
+	os.Exit(1)
+
 }
 
 // ListenAndServeKubeletReadOnlyServer initializes a server to respond to HTTP network requests on the Kubelet.
 func ListenAndServeKubeletReadOnlyServer(host HostInterface, resourceAnalyzer stats.ResourceAnalyzer, address net.IP, port uint, enableCAdvisorJSONEndpoints bool) {
-	klog.V(1).Infof("Starting to listen read-only on %s:%d", address, port)
```
Comment (yuzhiquan) [inline pkg/kubelet/server/server.go:170]: Same with line 148

File: pkg/kubelet/server/server.go [hunk 3]
```diff
@@ -272,13 +278,13 @@ func (s *Server) InstallAuthFilter() {
 		decision, _, err := s.auth.Authorize(req.Request.Context(), attrs)
 		if err != nil {
 			msg := fmt.Sprintf("Authorization error (user=%s, verb=%s, resource=%s, subresource=%s)", attrs.GetUser().GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			klog.Errorf(msg, err)
```
Comment (yuzhiquan) [inline pkg/kubelet/server/server.go:275]: May be 
```go
klog.ErrorS(err, "Authorization error", "user",attrs.GetUser().GetName(),"verb",attrs.GetVerb(),"resource",attrs.GetResource(),"subresource",attrs.GetSubresource())
```

File: pkg/kubelet/server/server.go [hunk 4]
```diff
@@ -407,7 +413,7 @@ const pprofBasePath = "/debug/pprof/"
 
 // InstallDebuggingHandlers registers the HTTP request patterns that serve logs or run commands/containers
 func (s *Server) InstallDebuggingHandlers() {
-	klog.Infof("Adding debug handlers to kubelet server.")
```
Comment (yuzhiquan) [inline pkg/kubelet/server/server.go:410]: Hmm.

File: pkg/kubelet/server/server.go [hunk 5]
```diff
@@ -272,13 +278,13 @@ func (s *Server) InstallAuthFilter() {
...
 		if err != nil {
 			msg := fmt.Sprintf("Authorization error (user=%s, verb=%s, resource=%s, subresource=%s)", attrs.GetUser().GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			klog.Errorf(msg, err)
+			klog.ErrorS(err, msg)
 			resp.WriteErrorString(http.StatusInternalServerError, msg)
 			return
 		}
 		if decision != authorizer.DecisionAllow {
 			msg := fmt.Sprintf("Forbidden (user=%s, verb=%s, resource=%s, subresource=%s)", attrs.GetUser().GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			klog.V(2).Info(msg)
```
Comment (yuzhiquan) [inline pkg/kubelet/server/server.go:281]: Same with above.

File: pkg/kubelet/server/server.go (hunk 1 above)
Comment (yxxhero) [reply pkg/kubelet/server/server.go:148]: Sorry, I don't understand your mean. `Starting to listen on 127.0.0.1=8080` is right? @yuzhiquan

File: pkg/kubelet/server/server.go (hunk 4 above)
Comment (yxxhero) [reply pkg/kubelet/server/server.go:410]: Hmm?????

File: pkg/kubelet/server/server.go (hunk 1 above)
Comment (yuzhiquan) [reply pkg/kubelet/server/server.go:148]: 127.0.0.1:8080 more properly.

File: pkg/kubelet/server/server.go (hunk 4 above)
Comment (yuzhiquan) [reply pkg/kubelet/server/server.go:410]: Not necessary.

File: pkg/kubelet/server/server.go (hunk 1 above)
Comment (yxxhero) [reply pkg/kubelet/server/server.go:148]: ok. I understand , I will change the code. thanks very much. My english is poor.

**Comment by adisky** [comment]
This PR covers whole pkg/kubelet/server https://github.com/kubernetes/kubernetes/pull/98643, so I think we do need this one.

**Comment by yxxhero** [comment]
oh. I know. I will close it. Thanks. @adisky

**Pull request closed without merging**